src/util/graphic.js

```diff
@@ -653,6 +653,10 @@ export function setLabelStyle(
         setTextStyle(emphasisStyle, emphasisModel, emphasisSpecified, opt, true);
     }
 
+    if (typeof normalModel.get('color') === 'function') {
+        normalStyle.textFill = normalModel.get('color')(labelDataIndex);
+    }
+
```

Comment:
(1) normalStyle.textFill should not be modified after `setTextStyle` called, because it will make the the style different from the cache created in `setTextStyle` for a feature that "auto add text stroke when text is inside a symbol".
(2) If we support `label.color` callback, we should better not call the callback at this view stage, but in visual stage, like `symbol` and `symbolSize` did. Otherwise, it bring trouble to further optimization or prevent other components from using this color in future without break change of calling time. 
(3) `labelDataIndex` should not pass to user directly, because it is a "inner index" that is not the raw data index when some data filter existing.

Reply:
Wow, so much thing I didn't think about, I will look into it